distribution/docs/src/main/resources/_contents/_catalog-contents/extending-catalog-contents.adoc

```diff
@@ -3916,6 +3916,57 @@ Does not handle multiple geometries yet.
 
 |===
 
+===== Create an XML Input Transformer using SaxEventHandlers [[saxEventHandlers]]
+
+If the transformer will transform XML (as opposed to JSON or a Word document, for example) there is a simpler solution than fully implementing a MetacardValidator.
+DDF includes an extensible, configurable `XmlInputTransformer`. This transformer can be instantiated via blueprint as a managed service factory and configured via metatype.
+The `XmlInputTransformer` takes a configuration of `SaxEventHandler`s.
+A `SaxEventHandler` is a class that handles SAX Events (a very fast XML parser) to parse metadata and create metacards.
+As many `SaxEventHandler`s as needed can be implemented and included in the `XmlInputTransformer` configuration. See the `catalog-transformer-streaming-impl` bundle for examples (`XmlSaxEventHandlerImpl` which parses the DDF Metacard XML Metadata and the `GmlHandler` which parses GML 2.0)
+Each `SaxEventHandler` implementation has a `SaxEventHandlerFactory` associated with it.
+The `SaxEventHandlerFactory` is responsible for instantiating new `SaxEventHandlers` - each transform request gets a new instance of `XmlInputTransformer` and set of `SaxEventHandler`s to be thread- and state-safe.
+
+The following diagrams intend to clarify implementation details:
+
+Figure 1 shows the XmlInputTransformer configuration, which is configured using the metatype and has the `SaxEventHandlerFactory` ids.
+Then, when a transform request is received, the `ManagedServiceFactory` instantiates a new `XmlInputTransformer`.
+This `XmlInputTransformer` then instantiates a new `SaxEventHandlerDelegate` with the configured `SaxEventHandlersFactory` ids.
+The factories all in turn instantiate a `SaxEventHandler`.
+Then, the `SaxEventHandlerDelegate` begins parsing the XML input document, handing the SAX Events off to each `SaxEventHandler`, which handle them if they can.
+After parsing is finished, each `SaxEventHandler` returns a list of `Attributes` to the `SaxEventHandlerDelegate` and `XmlInputTransformer` which add the attributes to the metacard and then return the fully constructed metacard.
+
+
+Fig. 1
+XmlInputTransformer Configuration
+|
+|
+--- String value of `SaxEventHandlerFactory1` id
+|
+|
+--- String value of `SaxEventHandlerFactory2` id
+
+Fig. 2
+XmlInputTransformer
+|
+--- SaxEventHandlerDelegate
+    |
+    |
+    --- `SaxEventHandlerFactory1` -> SaxEventHandler1
+    |
+    |
+    --- `SaxEventHandlerFactory2` -> SaxEventHandler2
+
```

Comment:
Add this line at the beginning of each diagram:
`[diagram-type, generated-file-name, generated-image-format]` (for example: `[ditaa, XmlInputTransformer_diagram, png]`)

and then put a `....` on the next line to start the diagram block
and another `....` after the diagram to close it. 

See also: http://asciidoctor.org/docs/asciidoctor-diagram/